images/metalk8s-keepalived/generate-config.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import subprocess
+import yaml
+
+import jinja2
+
+
+def get_interface_from_ip(ip):
+    # NOTE: We do not have any easy way (without an external lib)
+    # to retrieve the interface easily in Python
+    # So let's rely on `ip route get`
+    output = subprocess.check_output(f"ip route get {ip}", shell=True, text=True)
+
+    match = re.search(r"^.+ dev (?P<iface>.+) src .+$", output, re.MULTILINE)
+    return match.group("iface")
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", "--input", help="Data input file", required=True)
+    parser.add_argument("-t", "--template", help=" Data template file", required=True)
+    parser.add_argument(
+        "-o", "--output", help="Output file (default stdout)", default="/dev/stdout"
+    )
+    args = parser.parse_args()
+
+    with open(args.input, mode="r") as in_fd:
+        input_data = yaml.safe_load(in_fd)
```

Comment:
Can you document the expected `input_data` format ?

Reply:
Not in the "metalk8s documentation" since it will be internal, but I should add a readme with it, you're right :+1:

Reply:
Yes a readme next to it would be enough IMO

Reply:
Wondering if we shouldn't add versioning for this schema :thinking: 
In case we change it (during an upgrade), there will be a time when different versions of this keepalived container (hence, this script) will have to read the same version of the input config (likely the oldest one). Versioning can help us handle backward compatibility in a more straightforward manner, I think.

Reply:
Yes, it's possible

Reply:
Done https://github.com/scality/metalk8s/pull/3844/commits/71d97073d1f3b63d69db4a102ae6461e54f3bd1b